src/main/java/lt/vu/trip/service/trip/TripServiceImpl.java

```diff
@@ -1,24 +1,64 @@
 package lt.vu.trip.service.trip;
 
 import lt.vu.trip.entity.Trip;
+import lt.vu.trip.entity.TripParticipationStatus;
+import lt.vu.trip.entity.User;
 import lt.vu.trip.repository.TripRepository;
+import lt.vu.trip.service.user.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-
 @Component
 public class TripServiceImpl implements TripService {
 
 	@Autowired
 	private TripRepository repo;
 
-	public List<Trip> getAll() {
-		return repo.findAll();
+	@Autowired
+	private UserService userService;
```

Comment:
Here are all the methods that implement fetching different trip lists. All of them receive the parameters/filters. This service gets additional filters when needed (for example current user Id) and the retrieves the list of trips from repository